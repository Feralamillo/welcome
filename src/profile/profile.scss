@import 'src/styles/variables.scss';

.container {
  display: flex;
  align-items: center;
  justify-content: center;
  height: 100vh;
}

.user-profile {
  background-color: $white;
  border-radius: 10px;
  box-shadow: 0px 4px 4px rgba($black, 0.25);
  text-align: center;
  padding: 2rem;

  .user-image {
    height: 200px;
    width: 200px;
    border-radius: 50%;
  }

  .title {
    color: $primary;
  }

  .food {
    font-weight: bold;
    color: $primary;
  }
}
