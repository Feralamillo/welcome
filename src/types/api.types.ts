export interface SuccessResponse<T> {
  success: true;
  result: T;
}

export interface FailedResponse<T> {
  success: false;
  status: number;
  message?: string;
}

export type ApiResponse<T> = SuccessResponse<T> | FailedResponse<T>;
