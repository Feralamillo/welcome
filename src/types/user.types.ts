export interface User {
  firstName: string;
  lastName: string;
  img: string;
  food: string;
}
