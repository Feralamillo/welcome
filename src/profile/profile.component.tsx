import React from 'react';

import { User } from '../types/user.types';

interface Props {
  user: User;
}

export const ProfileComponent: React.FC<Props> = ({
  user: { firstName, lastName, img, food },
}) => (
  <div className='container'>
    <div className='user-profile'>
      <img src={img} alt={`${firstName} ${lastName}`} className='user-image' />
      <h1 className='title'>{`${firstName} ${lastName}`}</h1>
      <p>
        My favourite food is <span className='food'>{food}</span>
      </p>
    </div>
  </div>
);
