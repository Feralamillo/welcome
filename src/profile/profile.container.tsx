import React, { useEffect, useState } from 'react';

import { userApi } from '../api/user.api';
import { User } from '../types/user.types';
import { Loader } from '../ui-components/loader/loader';
import { ProfileComponent } from './profile.component';

import './profile.scss';

export const Profile: React.FC = () => {
  const [user, setUser] = useState<User | null>(null);
  const [isFetching, setFetching] = useState(true);

  const fetchUser = async () => {
    const response = await userApi.getUser();

    if (response.success) {
      setUser(response.result);
    } else {
      /* @TODO: error management */
      setUser(null);
    }

    setFetching(false);
  };

  useEffect(() => {
    fetchUser();
  }, []);

  if (isFetching) {
    return (
      <div className='container'>
        <Loader />
      </div>
    );
  }

  if (!user) {
    return (
      <div className='container'>
        <div>Sorry...there has been an error</div>
      </div>
    );
  }

  return <ProfileComponent user={user} />;
};
