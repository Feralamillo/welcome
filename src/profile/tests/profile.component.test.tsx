import { render } from '@testing-library/react';
import { User } from '../../types/user.types';
import { ProfileComponent } from '../profile.component';

describe('<ProfileComponent >', () => {
  it('should render the component', () => {
    const mockUser: User = {
      firstName: 'Jim',
      lastName: 'Morrison',
      food: 'Pizza',
      img: 'srcImage',
    };
    const { container } = render(<ProfileComponent user={mockUser} />);

    expect(container).toMatchSnapshot();
  });
});
