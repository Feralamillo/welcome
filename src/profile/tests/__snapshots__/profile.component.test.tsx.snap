// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProfileComponent > should render the component 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="user-profile"
    >
      <img
        alt="Jim Morrison"
        class="user-image"
        src="srcImage"
      />
      <h1
        class="title"
      >
        Jim Morrison
      </h1>
      <p>
        My favourite food is 
        <span
          class="food"
        >
          Pizza
        </span>
      </p>
    </div>
  </div>
</div>
`;
